activation/activation.go

```diff
@@ -875,11 +887,18 @@ func (b *Builder) searchPositioningAtx(
 		VerifyChainOpts.AssumeValidBefore(time.Now().Add(-b.postValidityDelay)),
 		VerifyChainOpts.WithTrustedID(nodeID),
 		VerifyChainOpts.WithLogger(b.logger),
+		VerifyChainOpts.PrioritizeCall(),
 	)
 	if err != nil {
-		logger.Info("search failed - using golden atx as positioning atx", zap.Error(err))
-		id = b.conf.GoldenATXID
+		if previous != nil {
+			id = previous.ID()
```

Comment:
The `id` must not be overwritten with `previous` here because we don't want to cache the previous ATX of a specific ID here. Note that the ATX builder runs for multiple IDs and it aims to prefer an ATX of a given ID over any other if it has equal tick height.

The point of returning the golden ATX from this function is to guarantee it always finds an ATX.

Reply:
Got it, thanks